feat: add /me routes to backend

bstojkovic opened the pull request:
This is a basic draft of a PATCH `/me` route. I am not sure if I am doing this correctly.

gikf reviewed (commented):
It looks correct, other than small nitpick in the suggestion.

Could you also add one more route - `/me/ideas` - listing all of the ideas of the current user?

bstojkovic commented:
> It looks correct, other than small nitpick in the suggestion.
> 
> Could you also add one more route - `/me/ideas` - listing all of the ideas of the current user?

I fixed it and added a draft of the /me/ideas route. I need a little help with how it should look like

bstojkovic commented on backend/src/api/routes/me.py:
```diff
@@ -0,0 +1,35 @@
+
+@router.get("/ideas", response_model=IdeasPublic)
+async def get_ideas(db: Db, current_user: Annotated[User, LoggedInUser]):
+    ideas = await db.find(Idea, creator_id=current_user.id)
```
I'm getting an error when I added your suggestion:
`Expected parameter name (Pylance)`
<img width="473" height="24" alt="image" src="https://github.com/user-attachments/assets/1394b2d0-a0f8-4136-a89d-2cd89e987630" />

gikf commented on backend/src/api/routes/me.py:
```diff
@@ -0,0 +1,41 @@
+
+@router.get("/")
+async def get_me(current_user: Annotated[User, LoggedInUser]):
+    return current_user
```
```suggestion
    return UserMe(**current_user.model_dump())
```
After this modification, the `"/me"` route in `main.py` can be completely removed.

gikf commented:
Hey @bstojkovic,

did you had a chance to look at the latest feedback? If you'd rather have me finalize these last touches - let me know.
